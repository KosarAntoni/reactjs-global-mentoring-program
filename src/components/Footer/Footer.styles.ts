import styled from '@emotion/styled'

export const StyledFooter = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;

  padding: 2.3rem;

  background-color: var(--gray);
`
