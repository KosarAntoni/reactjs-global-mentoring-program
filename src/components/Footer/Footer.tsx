import React, { FC } from 'react'

import Logo from 'components/Logo'

import './Footer.styles.scss'

const Footer: FC = () => {
  return (
    <footer className='footer' data-testid='footer-node'>
      <Logo/>
    </footer>
  )
}

export default Footer
