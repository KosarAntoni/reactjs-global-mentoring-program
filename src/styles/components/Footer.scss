.footer {
  display: flex;
  align-items: center;
  justify-content: center;

  padding: 2.3rem;

  background-color: var(--gray);
}
