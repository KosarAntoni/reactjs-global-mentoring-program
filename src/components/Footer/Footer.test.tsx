import React from 'react'
import { render, screen } from '@testing-library/react'

import Footer from './Footer'

import '@testing-library/jest-dom'

test('the Footer component renders correctly with required and optional props', () => {
  render(<Footer />)

  const node = screen.getByTestId('footer-node')

  // the component has been rendered
  expect(node).toBeInTheDocument()

  // generating a snapshot, not for testing purposes but only to see what changes in the HTML structure during CR if the developer updates the snapshot
  expect(node).toMatchSnapshot()

  // has Logo componrnt
  expect(screen.getByTestId('logo-node')).toBeInTheDocument()

  // the comonent has correct classes
  expect(node).toHaveClass('footer')
})
