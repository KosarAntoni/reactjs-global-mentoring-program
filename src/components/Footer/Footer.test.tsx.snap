// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Footer component renders correctly with required and optional props 1`] = `
<footer
  class="footer"
  data-testid="footer-node"
>
  <span
    class="logo"
    data-testid="logo-node"
  >
    <b>
      netflix
    </b>
    roulette
  </span>
</footer>
`;
